### PR TITLE
Updated cert type for swarm cert/key

### PR DESCRIPTION
**Pull request by tripdubroot**
PR for Issue #575 

Signed-off-by: Ian Philpot <ianphil@microsoft.com>

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/580)

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -76,6 +76,14 @@ type CertificateProfile struct {
 	KubeConfigPrivateKey string `json:"kubeConfigPrivateKey,omitempty"`
 	// caPrivateKey is an internal field only set if generation required
 	caPrivateKey string
+	// SwarmTlsServerCertificate is the Docker server daemon certificate and signed by the CA
```
Comment (amanohar) [inline pkg/api/types.go]: CertificateProfile also contains the following which is K8s specific:

```
// KubeConfigCertificate is the client certificate used for kubectl cli and signed by the CA
KubeConfigCertificate string `json:"kubeConfigCertificate,omitempty"`
// KubeConfigPrivateKey is the client private key used for kubectl cli and signed by the CA
KubeConfigPrivateKey string `json:"kubeConfigPrivateKey,omitempty"`
```
SwarmMode cert and Kube config certs live side by side in one struct? Its confusing about what is the right combination. Should the CertificateProfile be renamed to KubernetesCertificateProfile and another struct created called "SwarmModeCertificateProfile". That way there can be validation check that one only type is set in properties based on orchestrator profile.

Comment (tripdubroot) [inline pkg/api/types.go]: @amanohar I'll make this change.

**Comment by shrutir25** [comment]
@acs-bot test this please

**Comment by shrutir25** [comment]
@amanohar - how do we want to proceed with this PR ?

**Comment by tripdubroot** [comment]
closing PR... will open new

**Pull request closed without merging**